sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/metrics/TraceScope.kt

```diff
@@ -0,0 +1,38 @@
+package cash.z.ecc.android.sdk.internal.metrics
+
+import android.os.Build
+import androidx.tracing.Trace
+import cash.z.ecc.android.sdk.internal.Twig
+import kotlin.random.Random
+
+class TraceScope {
+    private val methodName: String
+    private var cookie: Int?
+
+    constructor(methodName: String) {
+        val cookie = Random.nextInt()
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            Trace.beginAsyncSection(methodName, cookie)
+        } else {
+            // Best-effort for older Android versions.
+            Trace.beginSection(methodName)
+        }
```

Comment:
For unknown reasons, I have not yet managed to get `Trace.beginAsyncSection` to show up anywhere. When I comment the async variants out and instead use the older `Trace.beginSection` etc. I get traces showing up (but not all correctly, due to `CompactBlockProcessor`'s heavy usage of `suspend`).

Reply:
Opened #1375